base/linalg/lapack.jl

```diff
@@ -664,7 +675,7 @@ for (gesvx, elty) in
             ferr      = similar(A, $elty, nrhs)
             berr      = similar(A, $elty, nrhs)
             work      = Array($elty, 4n)
-            iwork     = Array($elty, n)
+            iwork     = Array($elty, 2n)
```

Comment:
Is the segfault in the real driver or the complex drivers? In the real case, the iwork is indeed n and consistent with other functions that called, which also expect iwork of size n.